exercise-004-cars/server.go

```
@@ -0,0 +1,152 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"html/template"
+	"net/http"
+	"sort"
+	"strings"
+	"sync"
+	"time"
+)
+
+var joinT *template.Template
+var playT *template.Template
+
+var vehicleLock sync.RWMutex
+
+type Vehicle struct {
+	Name  string
+	Count int
+}
+
+type Vehicles struct {
+	List []*Vehicle
+}
+
+type View struct {
+	Username string
+	Vehicles Vehicles
+}
+
+var usernameVehiclesByCount = make(map[string]map[string]int)
+
+func handleJoin(w http.ResponseWriter, r *http.Request) {
+	username, _ := r.Cookie("username")
```

You shouldn't ignore errors unless:
1. The error can never occur, under any circumstance; or
2. If the error does occur, it's inconsequential to the rest of the application.

This isn't one of those cases, you should handle this error.